generator/src/main/java/com/scottlogic/deg/generator/validators/GenerationConfigValidator.java

```diff
@@ -8,80 +8,116 @@
-                errorMessages.add(
-                    "Invalid Output - target is a directory, please use a different output filename");
-            } else if (!configSource.overwriteOutputFiles() && fileUtils.exists((FileOutputTarget)outputTarget)) {
-                errorMessages.add(
-                    "Invalid Output - file already exists, please use a different output filename or use the --overwrite option");
-            }
+    private void checkSwitches (GenerationConfig config, ArrayList<String> errorMessages) {
+        if (config.getDataGenerationType() == GenerationConfig.DataGenerationType.RANDOM
+            && !config.getMaxRows().isPresent()) {
+
+            errorMessages.add("RANDOM mode requires max row limit: use -n=<row limit> option");
         }
     }
 
-    private void checkViolationGenerateOutputTarget(ArrayList<String> errorMessages,
-                                                    OutputTarget outputTarget, int ruleCount) {
-        if (outputTarget instanceof FileOutputTarget) {
-            if (!fileUtils.exists((FileOutputTarget)outputTarget)) {
-                errorMessages.add(
-                    "Invalid Output - output directory must exist, please enter a valid directory name");
-            } else if (!fileUtils.isDirectory((FileOutputTarget)outputTarget)) {
-                errorMessages
-                    .add("Invalid Output - not a directory, please enter a valid directory name");
-            } else if (!configSource.overwriteOutputFiles() && !fileUtils
-                .isDirectoryEmpty((FileOutputTarget)outputTarget, ruleCount)) {
-                errorMessages.add(
-                    "Invalid Output - directory not empty, please remove any 'manfiest.json' and '[0-9].csv' files or use the --overwrite option");
+    private void checkProfileInputFile(ArrayList<String> errorMessages, File profileFile) {
+        if (fileUtils.containsInvalidChars(profileFile)) {
+            errorMessages.add(String.format("Profile file path (%s) contains one or more invalid characters ? : %% \" | > < ", profileFile.toString()));
+        }
+        else if (!profileFile.exists()){
+            errorMessages.add("Invalid Input - Profile file does not exist");
+        }
+        else if (profileFile.isDirectory()) {
+            errorMessages.add("Invalid Input - Profile file path provided is to a directory");
+        }
+        else if (fileUtils.isFileEmpty(profileFile)) {
+            errorMessages.add("Invalid Input - Profile file has no content");
+        }
+        else {
+            try {
+                String fileType = fileUtils.probeContentType(profileFile.toPath());
+                if (!fileType.equals("application/json")) {
+                    errorMessages.add(String.format("Invalid Input - File is of type %s\nFile type application/json required", fileType));
+                }
+            } catch (IOException e) {
+                e.printStackTrace();
```

Comment:
There is no need for this, the detail will be in the validation message. Also it would emit to system.err where it should be printed to a logging framework

Reply:
What should we put into the catch block ? or should we just throw the exception?

Reply:
or should we consume the exception inside `fileUtils.probeContentType` and return an `unknown` type?

Reply:
we're adding a validation failure to the `errorMessages` collection, therefore the detail is captured, i'd say swallow the exception.

In future when there is a logging framework in place, we could do a VERBOSE/DEBUG log here.

Reply:
My comment was over the call to `e.printStackTrace();` only, the `catch` is required.

Reply:
The ProfileReader would catch this scenario and throw an exception, concern is that this process could report an error that the ProfileReader could accept